deepspeed/pt/deepspeed_light.py

```diff
@@ -533,8 +542,11 @@ def _configure_basic_optimizer(self, model_parameters):
                 "'max_grad_norm' is not supported as an optimizer parameter, please switch to using the deepspeed parameter 'gradient_clipping' see: https://www.deepspeed.ai/docs/config-json/#gradient-clipping for more details"
             )
         if self.optimizer_name() == ADAM_OPTIMIZER:
-            from apex.optimizers.fused_adam import FusedAdam
-            optimizer = FusedAdam(model_parameters, **optimizer_parameters)
+            if self.zero_cpu_offload():
```

Comment:
I don't believe this code path can be reached because we would have already triggered the `zero_cpu_offload` condition in `_configure_optimizer()`.

Reply:
yes the zero_cpu_offload condition will in configure_optimizer(). I can delete the check in _configure_basic_optimizer